benchmarks/cpu_haswell.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#SBATCH --nodes=1
+#SBATCH --constraint=haswell
+#SBATCH --time=15
+#SBATCH --qos=debug
+
+echo "Start time: $(date --iso-8601=seconds)"
+
+# Returns the number of seconds since unix epoch
+_time() {
+    echo $(date +%s.%N)
+}
+
+# Returns the elapsed time in seconds between start and end
+# setting scale=3 and dividing by 1 to limit to millisecond precision
+_elapsed_time () {
+    echo "scale=3; ($2 - $1) / 1" | bc
+}
+
+# Environement setup
+source /global/cfs/cdirs/desi/software/desi_environment.sh master
```

Comment:
Since our environment isn't necessarily happy about being setup twice, consider protecting this with something like
```
if [ -z "$DESIMODULES" ]; then
    source /global/cfs/cdirs/desi/software/desi_environment.sh master
fi
```
Checking: I think we only need this to get an anaconda environment that has fitsio; gpu_specter itself doesn't require any of the desi specific packages, but our desiconda environment is a convenient way to get the external packages we need.

Reply:
For the cpu baseline version it's probably best to benchmark with the real desi environment since it is arguably the most "fair" measurement. For the gpu version, we'll have to switch over to our own custom conda environment. Just something to keep in mind.

Reply:
Good point @lastephey.  I think that implies that the benchmark script should *not* source an environment, but rather require that the user has done that ahead of time.  If desired, it could still check that it has the basic pieces it needs (like import fitsio, import gpu_specter) and exit with a helpful message rather than proceeding to crash spex.